src/Promitor.Core.Scraping/LogAnalyticsScraper.cs

```diff
@@ -0,0 +1,54 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using GuardNet;
+using Microsoft.Azure.Management.Monitor.Fluent.Models;
+using Promitor.Core.Contracts;
+using Promitor.Core.Contracts.ResourceTypes;
+using Promitor.Core.Metrics;
+using Promitor.Core.Scraping.Configuration.Model.Metrics;
+using Promitor.Integrations.LogAnalytics;
+
+namespace Promitor.Core.Scraping
+{
+    public class LogAnalyticsScraper : Scraper<LogAnalyticsResourceDefinition>
+    {
+        private const string ResourceUriTemplate = "subscriptions/{0}/resourceGroups/{1}/providers/Microsoft.OperationalInsights/workspaces/{2}/Overview";
+        private readonly LogAnalyticsClient _logAnalyticsClient;
+
+        public LogAnalyticsScraper(ScraperConfiguration scraperConfiguration) : base(scraperConfiguration)
+        {
+            _logAnalyticsClient = scraperConfiguration.LogAnalyticsClient;
+        }
+
+        protected override async Task<ScrapeResult> ScrapeResourceAsync(string subscriptionId, ScrapeDefinition<IAzureResourceDefinition> scrapeDefinition, LogAnalyticsResourceDefinition resourceDefinition, AggregationType aggregationType, TimeSpan aggregationInterval)
+        {
+            Guard.NotNull(scrapeDefinition, nameof(scrapeDefinition));
+            Guard.NotNull(scrapeDefinition.LogAnalyticsConfiguration, nameof(scrapeDefinition.LogAnalyticsConfiguration));
+
+            string resourceUri = BuildResourceUri(subscriptionId, scrapeDefinition, resourceDefinition);
+            var query = scrapeDefinition.LogAnalyticsConfiguration.Query;
+            var workspaceId = resourceDefinition.WorkspaceId;
+
+            List<MeasuredMetric> measuredMetrics = new List<MeasuredMetric>();
+
+            // Query Azure Log Analytics for result
+            var result = await _logAnalyticsClient.RunKustoQueryAsync(workspaceId, query, aggregationInterval);
+            var measuredMetric = MeasuredMetric.CreateWithoutDimension(result);
+            measuredMetrics.Add(measuredMetric);
+            var metricLabels = DetermineMetricLabels(resourceDefinition);
```

Comment:
```suggestion
            var metricLabels = DetermineMetricLabels(resourceDefinition);
            var measuredMetric = MeasuredMetric.Create(result, metricLabels);
            measuredMetrics.Add(measuredMetric);
```

Reply:
This is teh one I was looking for: https://github.com/tomkerkhove/promitor/blob/e1fdb945a79a0a58b20411f7288ef98cc8c1e6f7/src/Promitor.Core/Metrics/MeasuredMetric.cs#L62-L78

Reply:
But CreateForDimension is the function with input `metrics` and `dimension`. But what we have here is `metrics` and `labels`

Reply:
A label is the same thing as a dimension; label is just the way Prometheus calls them, dimension is how Azure Monitor calls it but it's OK.

It works, it's just an optimization and will resolve

Reply:
Ok. But it seem like `labels` and `dimension` can not be duplicated. We already have 2 labels `workspace_id` and `workspace_name` here. If I do something like 
```
var measuredMetric = MeasuredMetric.CreateForDimension(result, "workspace_id", "some workpsace id");
measuredMetrics.Add(measuredMetric);
```
then I got this exception 
```
      System.ArgumentException: An item with the same key has already been added. Key: workspace_id
         at System.Collections.Generic.Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior)
         at System.Collections.Generic.Dictionary`2.Add(TKey key, TValue value)
         at Promitor.Integrations.Sinks.Prometheus.PrometheusScrapingEndpointMetricSink.DetermineLabels(PrometheusMetricDefinition metricDefinition, ScrapeResult scrapeResult, MeasuredMetric measuredMetric, Dictionary`2 defaultLabels) in /Users/ltran/DataPlatform/promitor/src/Promitor.Integrations.Sinks.Prometheus/PrometheusScrapingEndpointMetricSink.cs:line 97
```

We already add labels for them. So I think no need to add dimension (from labels) anymore

Reply:
That's not the intention but rather pass the dimensions from the metric but this is not Azure Monitor so that does not make sense - Ignore me :)